### PR TITLE
Settings fixing

### PR DESCRIPTION
**Pull request by YAWAL**
functionalities for blocking/unblocking of contacts, checking if contact is blocked, changing password and nickname have been fixed and adopted to our DB

### CODE REVIEW

File: settingService/blockUser.go [hunk 1]
```diff
@@ -1,54 +1,55 @@
...
-		return nil, -1, err
+		return "", -1, err
 	}
-	IsBlocked := body.IsBlocked // IsBlocked value could be int 0 or 1
-	if IsBlocked != 0 && IsBlocked != 1 {
-		err := errors.New("IsBlocked value is not valid. IsBlocked = " + string(IsBlocked))
+	contactUser = body.ContactUser
+	isBlocked = body.IsBlocked // IsBlocked value could be int 0 or 1
+	if isBlocked != 0 && isBlocked != 1 {
+		err := errors.New("IsBlocked value is not valid. IsBlocked = " + string(isBlocked))
```
Comment (8tomat8) [inline settingService/blockUser.go]: Excess text

File: settingService/blockUser.go [hunk 2]
```diff
@@ -1,54 +1,55 @@
...
 		loger.Log.Errorf("IsBlocked value is not 1 or 0:", err)
-		return nil, -1, err
+		return "", -1, err
 	}
-	return body, IsBlocked, nil
+	if contactUser == "" {
+		err := errors.New("ContactUser value is empty")
+		loger.Log.Errorf("Error has occurred: ", err)
```
Comment (8tomat8) [inline settingService/blockUser.go]: Same as previous

File: settingService/blockUser.go [hunk 3]
```diff
@@ -57,47 +58,35 @@ func BlockUnblockUser(request *messageService.Message) (bool, error) {
...
 	if err != nil {
 		loger.Log.Errorf("Error has occurred: ", err)
 		return true, err
 	}
 	mainUser := body.ReceiverName
 	contactUser := request.Header.UserName
-	db, err := database.GetStorage() // common gorm-connection from database package
+	db, err := database.GetStorage()
 	if err != nil {
 		loger.Log.Errorf("DB error has occurred: ", err)
```
Comment (8tomat8) [inline settingService/blockUser.go]: remove `f`

File: settingService/blockUser.go [hunk 4]
```diff
@@ -57,47 +58,35 @@ func BlockUnblockUser(request *messageService.Message) (bool, error) {
...
 func IsUserBlocked(request *messageService.Message) (isBlocked bool, err error) {
 	isBlocked = true
-	var body messageService.MessageBody
+	var body *messageService.MessageBody
 	err = json.Unmarshal(request.Body, &body)
 	if err != nil {
 		loger.Log.Errorf("Error has occurred: ", err)
 		return true, err
 	}
 	mainUser := body.ReceiverName
```
Comment (8tomat8) [inline settingService/blockUser.go]: Try not to copy data if you don't need it

File: settingService/changePassNickName.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	NickName string `json:"nick_name"`
+}
+
+// UnmarshalNickNameRequest function retrieves values of password and nickName from request body
+func UnmarshalRequestBody(request *messageService.Message) (pass string, nickName string, err error) {
+	var body *ChangeNickNamePassRequestBody
+	err = json.Unmarshal(request.Body, &body)
+	if err != nil {
+		loger.Log.Errorf("Error has occurred: ", err)
+		return "", "", err
```
Comment (8tomat8) [inline settingService/changePassNickName.go]: You can call return without any params

File: settingService/changePassNickName.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+	"encoding/json"
+	"errors"
+
+	"github.com/8tomat8/SSU-Golang-252-Chat/database"
+	"github.com/8tomat8/SSU-Golang-252-Chat/loger"
+	"github.com/8tomat8/SSU-Golang-252-Chat/messageService"
+)
+
+// ChangeNickNamePassRequestBody is a custom body for request to change nick_name or password
+type ChangeNickNamePassRequestBody struct {
```
Comment (8tomat8) [inline settingService/changePassNickName.go]: Find better name

File: settingService/changePassNickName.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+		return "", err
+	}
+	return userName, nil
+}
+
+// ChangeNickName perform changing users NickName value in authentifications table.
+func ChangeNickName(request *messageService.Message) (bool, error) {
+	userName, err := GetUserName(request)
+	if err != nil {
+		loger.Log.Errorf("Error has occurred: ", err)
```
Comment (8tomat8) [inline settingService/changePassNickName.go]: Duplicated logging

**Review by 8tomat8 — APPROVED** [review]

File: settingService/blockUser.go (hunk 1 above)
Comment (YAWAL) [reply settingService/blockUser.go]: fixed: excessed text has been deleted

File: settingService/blockUser.go (hunk 2 above)
Comment (YAWAL) [reply settingService/blockUser.go]: fixed: excessed text has been deleted

File: settingService/blockUser.go (hunk 4 above)
Comment (YAWAL) [reply settingService/blockUser.go]: fixed: code has been refactored

File: settingService/blockUser.go (hunk 3 above)
Comment (YAWAL) [reply settingService/blockUser.go]: fixed

File: settingService/changePassNickName.go (hunk 6 above)
Comment (YAWAL) [reply settingService/changePassNickName.go]: fixed: structure has been renamed

File: settingService/changePassNickName.go (hunk 5 above)
Comment (YAWAL) [reply settingService/changePassNickName.go]: fixed: function has been refactored

File: settingService/changePassNickName.go (hunk 7 above)
Comment (YAWAL) [reply settingService/changePassNickName.go]: fixed: duplicated logging has been removed

**Pull request merged**